### DISCUSSION TITLE
Migrating all data to a new setup

### DISCUSSION THREAD

**Discussion by prmbittencourt** [firefly-iii/firefly-iii#9319] [Firefly III]
Hello. Currently I have a Firefly III docker image running on a fly.io server. I wish to deactivate it and move to a local instance on my home computer. I would like to know how to migrate my transactions, accounts, etc to my new local instance. Basically, I want to install Firefly III on my local machine and have it contain all the same accounts and data as the instance currently in the cloud, which I'm going to delete.

I know pretty much nothing about Docker other than that I deeply dislike everything about it and honestly it's a miracle that I even got my fly.io instance working at all, but I understand that the Firefly III image connects to a separate image running MariaDB, and that this image may be the one I need to get the data from. But maybe not, and I don't even know how to go about finding out which one it is, so I am throwing myself at the community's feet and begging for help. :)

I have exported my transactions to a CSV file following the link in the sidebar, but I'm pretty sure that's not enough, and I also don't know how I'd import all that data back into the new Firefly III instance.

Thank you for your time.

**Comment by JC5** [comment]
The easiest way would be to use something like mysqldump on the old server to export the data, and use the same tool to import it into the new one.

**Reply by prmbittencourt** [reply]
Thank you very much. Pardon my ignorance, but I have two questions that I couldn't find the answer for in the documentation.

1) Do I point mysqldump at the container running Firefly III, or do I need to expose the MariaDB container to the Internet and connect directly to it?

2) Do I need to install mysqldump in the target container as well as my local machine?
